azure-pipelines-dependency-graph-testing.yml

```diff
@@ -148,10 +148,9 @@ stages:
 
           - script: |
               pip install pytest pytest-cov pytest-azurepipelines
-              echo "Tests identified by dependency graph test runner [dgtr]:"
-              python scripts/determine_tests_to_run.py --depth 3 --ignore 'tests/cli' 'tests/integration/usage_statistics' | tee .dgtr
-              if [ -s .dgtr ]; then
-                cat .dgtr | xargs pytest $(GE_pytest_opts) --napoleon-docstrings --junitxml=junit/test-results.xml --cov=. --cov-report=xml --cov-report=html
+              test_files=$(python scripts/determine_tests_to_run.py --depth 3 --ignore 'tests/cli' 'tests/integration/usage_statistics')
+              if [ $? -eq 0 ]; then
+                echo $test_files | xargs pytest $(GE_pytest_opts) --napoleon-docstrings --junitxml=junit/test-results.xml --cov=. --cov-report=xml --cov-report=html
```

Comment:
Rather than write to a file, save the script output in a var and use the status code to conditionally run pytest